packages/php-wasm/compile/build-assets/Makefile

```diff
@@ -0,0 +1,221 @@
+PKG_CONFIG_PATH=/root/lib/lib/pkgconfig
+TIMER=(which pv > /dev/null && pv --name '${@}' || cat)
+
+.PHONY: all
+all: build
+
+.PHONY: build
+build: libz.a libzip.a libpng.a libxml2.a libsqlite3.a libbison.a
+	$(TIMER) && echo "Done building"
+
+.PHONY: clean
+clean:
+	docker rmi my-image
+
+.PHONY: run
+run:
+	docker run my-image
+
+.PHONY: shell
+shell:
+	docker run -it my-image /bin/bash
+
+libz.a:
+	chmod a+x zlib/configure;
+	cd zlib && emconfigure ./configure --prefix=/root/lib
+	cd zlib && (EMCC_FLAGS="-fPIC" emmake make || true)
+	cd zlib && EMCC_FLAGS="-fPIC" emmake make install
+	touch libz.a
```

Comment:
this target is a little odd. we're designating a specific file, but then not creating it. we're `touch`-ing the file so `make` won't re-run the rule, but where is the real `libz.a` and why not rely on it?

`make` works best when expressed as dependencies on files or build artifacts, but I think this is expressing a task, which it is notoriously bad at expressing.

in one exploration I did with Calypso I created [the `$~` helper](https://github.com/Automattic/wp-calypso/pull/23225/files#diff-472ef2097641f6da91ae1da2de1533640b40cb169e80ae9c7cfd3d0076e401e2R70-R80) which can be used at the end of a rule to generate a sentinel file for that rule in a similar way you are doing here.

I don't understand this well enough to give more helpful commentary.

 - why are we using `|| true` and continuing to install if the `emmake make` step failed?
 - each line spawns a new shell, but if we continue the lines it'll happen in one
 - what are we trying to express with this target/recipe? what needs to be built? what is the significance of `libz.a` here?

```Makefile
libz.a:
	chmod a+x zlib/configure; && \
	cd zlib && \
	emconfigure ./configure --prefix=/root/lib && \
	(EMCC_FLAGS="-fPIC" emmake make || true) && \
	(EMCC_FLAGS="-fPIC" emmake make install)
	$~
```